src/main/java/seedu/address/logic/parser/InternshipSortCommandParser.java

```diff
@@ -0,0 +1,184 @@
+                    return field;
+                }
+            }
+            throw new ParseException(InternshipSortCommand.MESSAGE_NO_FIELD);
+        }
+    }
+
+    /** Enum of order to sort by */
+    public enum OrderEnum {
+        ASCENDING(ORDER_ASCENDING), DESCENDING(ORDER_DESCENDING);
+        private final String value;
+
+        OrderEnum(String order) {
+            this.value = order;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        /**
+         * Returns the {@code OrderEnum} based on the given order.
+         * @param order order to sort by
+         * @return the {@code OrderEnum} based on the given order
+         */
+        public static OrderEnum getOrderEnum(String order) throws ParseException {
+            requireNonNull(order);
+            isValidOrder(order);
+            if (order.equals(ORDER_ASCENDING)) {
+                return ASCENDING;
+            } else {
+                return DESCENDING;
+            }
+        }
+
+        /**
+         * Returns true if the given order is valid.
+         * @param trimmedOrder order to sort by
+         * @return true if the given order is valid
+         */
+        public static boolean isValidOrder(String trimmedOrder) throws ParseException {
+            requireNonNull(trimmedOrder);
+            if (trimmedOrder.equals(ORDER_ASCENDING) || trimmedOrder.equals(ORDER_DESCENDING)) {
+                return true;
+            } else {
+                throw new ParseException(InternshipSortCommand.MESSAGE_INVALID_ORDER);
+            }
+        }
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the InternshipSortCommand
+     * and returns an InternshipSortCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public InternshipSortCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        logger.info("Parsing internship sort command with args: " + args);
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            logger.warning("Internship sort command has no arguments");
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, InternshipSortCommand.MESSAGE_USAGE));
+        }
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, SUPPORTED_PREFIXES);
+        // is there a way to check which prefix was added into argmultimap?
+
+        if (!anyPrefixesPresent(argMultimap, SUPPORTED_PREFIXES)) {
+            logger.warning("Internship sort command has no valid prefixes");
+            throw new ParseException(InternshipSortCommand.MESSAGE_INVALID_FIELD);
+        }
+        OrderEnum parsedOrder = assignOrder(argMultimap);
+        FieldEnum field = assignField(argMultimap);
+        return new InternshipSortCommand(field, parsedOrder);
```

Comment:
if multiple prefixes are present, i assume only the first one is taken while the rest is treated as the string for the first prefix (and therefore invalid_order is thrown)? also based on 145-150 where assignField returns after finding the first prefix

Reply:
Yes this is right